packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -577,6 +578,29 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
 
     return tempPlist;
   }
+
+  // As of Xcode 15.4, the old export methods 'app-store', 'ad-hoc', and 'development'
+  // are now deprecated. The new equivalents are 'app-store-connect', 'release-testing',
+  // and 'debugging'.
+  String _getVersionAppropriateExportMethod(String method) {
+    final Version? currVersion = globals.xcode!.currentVersion;
+    if (currVersion != null) {
+      if (currVersion.major >= 16 || (currVersion.major == 15 && currVersion.minor > 3)) {
+        switch (method) {
+          case 'app-store':
+            return 'app-store-connect';
+          case 'ad-hoc':
+            return 'release-testing';
```

Comment:
Can you add tests for this?
https://github.com/flutter/flutter/blob/d4cffa25402d9a26d4eeee7704e69b0cda4a7da8/packages/flutter_tools/test/commands.shard/hermetic/build_ipa_test.dart#L618-L619

Reply:
done